packages/lib/src/index.ts

```diff
@@ -99,17 +113,27 @@ export default function federation(
       if (!Array.isArray(_options.external)) {
         _options.external = [_options.external as string]
       }
-      for (const pluginHook of pluginList) {
-        pluginHook.options?.call(this, _options)
+      for (const pluginHook of pluginList) {(
+            pluginHook.options as ((
+              this: MinimalPluginContext,
+              options: InputOptions
+            ) => InputOptions | NullValue)|undefined
+          )?.call(this, _options)
       }
       return _options
     },
     config(config: UserConfig, env: ConfigEnv) {
       options.mode = options.mode ?? env.mode
       registerPlugins(options.mode, env.command)
       registerCount++
-      for (const pluginHook of pluginList) {
-        pluginHook.config?.call(this, config, env)
+      for (const pluginHook of pluginList) {(
+            pluginHook.config as ((
```

Comment:
May feel cleaner to split this into two calls:
1. Get a type-safe config object
2. Invoke `.call()` on that object